icechunk/src/format/snapshot.rs

```diff
@@ -87,35 +64,83 @@ impl ZarrArrayMetadata {
     ///
     /// Returns false if the chunk coordinates are invalid.
     pub fn valid_chunk_coord(&self, coord: &ChunkIndices) -> bool {
-        debug_assert_eq!(self.shape.len(), coord.0.len());
-
         coord
             .0
             .iter()
-            .zip(self.max_chunk_indices_permitted().0)
+            .zip(self.max_chunk_indices_permitted())
             .all(|(index, index_permitted)| *index <= index_permitted)
     }
+
+    /// Returns an iterator over the maximum permitted chunk indices for the array.
+    ///
+    /// This function calculates the maximum chunk indices based on the shape of the array
+    /// and the chunk shape, using (shape - 1) / chunk_shape. Given integer division is truncating,
+    /// this will always result in proper indices at the boundaries.
+    fn max_chunk_indices_permitted(&self) -> impl Iterator<Item = u32> + '_ {
+        self.0.iter().map(|dim_shape| {
+            if dim_shape.chunk_length == 0 || dim_shape.array_length == 0 {
+                0
+            } else {
+                ((dim_shape.array_length - 1) / dim_shape.chunk_length) as u32
+            }
+        })
+    }
 }
 
-#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub enum DimensionName {
```

Comment:
Just for understanding: why pick this over `Option`?

Reply:
I just thought it was more clear ... DimensionName not set has a significant meaning, it's not just that they didn't pass it, they intentionally skipped it.